assets/js/components/CurrentUser.js

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import gql from "graphql-tag";
+import { useQuery } from "@apollo/react-hooks";
+import Error from "./UI/Error";
+
+const GET_CURRENT_USER_QUERY = gql`
+  query GetCurrentUser {
+    me {
+      username
+      email
+      displayName
+    }
+  }
+`;
+
+function CurrentUser({ children }) {
+  const { loading, error, data } = useQuery(GET_CURRENT_USER_QUERY);
+
+  if (error) return <Error error={error} />;
+
+  return children(data.me);
+}
+
+CurrentUser.propTypes = {
+  children: PropTypes.func.isRequired
+};
+
+export default CurrentUser;
+export { GET_CURRENT_USER_QUERY };
```

Comment:
What about creating a `/assets/js/components/Auth` directory, where we can organize Auth related components (`CurrentUser`, `RequireSignIn`) and queries?  Then we could create `.../Auth/auth.query.js` or something and avoid exporting from individual components?

Reply:
Yeah. I think that's a good idea. Some of that stuff didn't seem to fit anywhere so I just threw it somewhere. Let me see if BMQ and MBK have comments and I'll change at same time.